core/app/models/spree/order.rb

```diff
@@ -27,6 +27,25 @@ def generate_number(options = {})
     alias :display_ship_total :display_shipment_total
     alias_attribute :ship_total, :shipment_total
 
+    MONEY_THRESHOLD  = 999_999
```

Comment:
I find a 999_999 value to be far too low.  When dealing with some crap currencies their values are thousands per dollar so while 1 mil sounds okay for the dollar and euro as a constraint other currencies will shit the bed on a thousand dollar order.